### PR TITLE
main: add fallback mechanism for LLVM commands

### LINKED ISSUE #305: TinyGo can't find wasm-ld
Running latest TinyGo from `dev`, with all dependencies installed, I get the following error compiling some wasm:

```
tinygo build -o ./html/test.wasm -target wasm --no-debug ./js/main.go
error: failed to link /tmp/tinygo332217407/main: exec: "wasm-ld-8": executable file not found in $PATH
make: *** [Makefile:10: tinygo] Error 1
```

After symlinking `/usr/bin/wasm-ld` to `/usr/bin/wasm-ld-8` it works. I think we should support both paths.

### PR DESCRIPTION
**Pull request by aykevl**
On Debian, all LLVM commands have a version suffix (clang-8, ld.lld-8,
wasm-ld-8, etc.). However. Most other distributions only provide a
version prefix for Clang and not for all the other commands.

This commit fixes the issue by trying the command with the version
suffix first and falling back to one without if needed.

fixes #305.

### CODE REVIEW

**Comment by johanbrandhorst** [comment]
Works for me!

File: linker-external.go [hunk 1]
```diff
@@ -14,8 +14,12 @@ import (
 //
 // This version always runs the linker as an external command.
 func Link(linker string, flags ...string) error {
-	cmd := exec.Command(linker, flags...)
-	cmd.Stdout = os.Stdout
-	cmd.Stderr = os.Stderr
-	return cmd.Run()
+	if cmdNames, ok := commands[linker]; ok {
+		return execCommand(cmdNames, flags...)
+	} else {
```
Comment (johanbrandhorst) [inline linker-external.go]: This else is superfluous

File: linker-builtin.go [hunk 2]
```diff
@@ -57,9 +57,13 @@ func Link(linker string, flags ...string) error {
 		return nil
 	default:
 		// Fall back to external command.
-		cmd := exec.Command(linker, flags...)
-		cmd.Stdout = os.Stdout
-		cmd.Stderr = os.Stderr
-		return cmd.Run()
+		if cmdNames, ok := commands[linker]; ok {
+			return execCommand(cmdNames, flags...)
+		} else {
```
Comment (johanbrandhorst) [inline linker-builtin.go]: This else is superfluous

File: linker-external.go (hunk 1 above)
Comment (aykevl) [reply linker-external.go]: It is not when linking with the system compiler.

https://github.com/tinygo-org/tinygo/blob/fa5df4f524c3b4f15360c2da964932692e3ab4af/target.go#L215

File: linker-builtin.go (hunk 2 above)
Comment (aykevl) [reply linker-builtin.go]: (same here)

File: linker-external.go (hunk 1 above)
Comment (johanbrandhorst) [reply linker-external.go]: I'm confused. Whats wrong with this?
```suggestion
if cmdNames, ok := commands[linker]; ok {
	return execCommand(cmdNames, flags...)
}
cmd := exec.Command(linker, flags...)
cmd.Stdout = os.Stdout
cmd.Stderr = os.Stderr
return cmd.Run()
```

Comment (aykevl) [reply linker-external.go]: Oh, a formatting issue (not a logic issue). Yes, I'll apply this.

**Comment by johanbrandhorst** [comment]
Is this ready to go?

**Comment by justinclift** [comment]
Excellent.  This should improve the initial experience for other CentOS 7 users too:

```
$ ls -l clang* lld* wasm-ld*
lrwxrwxrwx. 1 jc jc        7 Apr 29 18:54 clang -> clang-8
lrwxrwxrwx. 1 jc jc        5 Apr 29 18:58 clang++ -> clang
-rwxr-xr-x. 1 jc jc 77273648 Apr 29 18:19 clang-8
-rwxr-xr-x. 1 jc jc 53606744 Apr 29 18:19 clang-check
lrwxrwxrwx. 1 jc jc        5 Apr 29 18:58 clang-cl -> clang
lrwxrwxrwx. 1 jc jc        5 Apr 29 18:58 clang-cpp -> clang
-rwxr-xr-x. 1 jc jc 21002424 Apr 29 18:19 clang-extdef-mapping
-rwxr-xr-x. 1 jc jc  2129232 Apr 29 18:18 clang-format
-rwxr-xr-x. 1 jc jc 26196048 Apr 29 18:19 clang-import-test
-rwxr-xr-x. 1 jc jc  3350200 Apr 29 18:18 clang-offload-bundler
-rwxr-xr-x. 1 jc jc 22313248 Apr 29 18:19 clang-refactor
-rwxr-xr-x. 1 jc jc 21435288 Apr 29 18:19 clang-rename
-rwxr-xr-x. 1 jc jc 42768232 Apr 29 18:21 lld
lrwxrwxrwx. 1 jc jc        3 Apr 29 18:58 lld-link -> lld
lrwxrwxrwx. 1 jc jc        3 Apr 29 18:58 wasm-ld -> lld
```

**Comment by deadprogram** [comment]
Great! Now merging.

**Pull request merged**